### PR TITLE
Initial schedule features

### PR DESCRIPTION
**Pull request by cretz**
## What was changed

* Added `schedule/cron` feature in all languages w/ basic cron behavior
* Added `schedule/basic_workflow` for just Go (no other released SDK has scheduling yet)

### CODE REVIEW

File: features/schedule/basic_workflow/feature.go
Comment (mjameswh) [inline features/schedule/basic_workflow/feature.go:33]: Issue has been fixed, this need to be a pointer.

File: features/schedule/basic_workflow/feature.go
Comment (mjameswh) [inline features/schedule/basic_workflow/feature.go:93]: I would use `DescribeScedule.info.recentActions` instead of listing all workflow executions. Would be much more easier and that would test a little bit more the schedule implementation itself.

File: features/schedule/basic_workflow/feature.go
Comment (mjameswh) [inline features/schedule/basic_workflow/feature.go:31]: I see more risks of bugs across SDKs with the structured calendar format than the interval spec format... Can we consider defining this test using an equivalent structured calandar spec format instead? For example (using TS syntax): `second: { start: 0, end: 59, step: 2 }, minute: '*', hour: '*', ...`

File: features/schedule/basic_workflow/feature.go
Comment (mjameswh) [inline features/schedule/basic_workflow/feature.go:28]: Starting a workflow every 2 seconds can be pretty aggressive in a "non production level" test environment. Just in case something goes wrong and the schedule stays active longer than expected, I would set `limitedActions = true`, and `remainingActions = 2`, or `endAt = now + 2 minutes`...

File: features/schedule/basic_workflow/feature.go
Comment (mjameswh) [inline features/schedule/basic_workflow/feature.go:30]: nit: I'd suggest following the test harness' naming convention for schedule id (`${this.source.relDir}-${randomUUID()}`), and `${scheduleId}-workflow` for the Workflow ID.

**Review by mjameswh — APPROVED** [review]
A few suggestions but overall LGTM

File: features/schedule/basic_workflow/feature.go
Comment (cretz) [reply features/schedule/basic_workflow/feature.go:29]: It is not resolved in a released version though. Cannot update.

File: features/schedule/basic_workflow/feature.go
Comment (cretz) [reply features/schedule/basic_workflow/feature.go:33]: It is not resolved in a released version though. Cannot update.

File: features/schedule/basic_workflow/feature.go
Comment (cretz) [reply features/schedule/basic_workflow/feature.go:30]: > following the test harness' naming convention for schedule id 

The harness doesn't have a naming convention for schedule ID. I don't think it matters much the shape of the identifier in these cases so long as unique. Notice the workflow type is the function name not prefixed with its qualified path to help the UI either.

File: features/schedule/basic_workflow/feature.go
Comment (cretz) [reply features/schedule/basic_workflow/feature.go:31]: That sounds like a new test. I can think of dozens and dozens of tests if we need to test each input form. These are the initial basic tests, not meant to validate each field.

File: features/schedule/basic_workflow/feature.go
Comment (cretz) [reply features/schedule/basic_workflow/feature.go:93]: I can add that. I still want to list workflows because I want to test the non-schedule result of the schedule on the actual system too.

File: features/schedule/basic_workflow/feature.go
Comment (cretz) [reply features/schedule/basic_workflow/feature.go:28]: The schedule is only supposed to be valid for the life of this test in an isolated way. I delete the schedule before this returns.

**Comment by Quinn-With-Two-Ns** [comment]
LGTM, assuming there will be another PR once typescript is released?

**Review by Quinn-With-Two-Ns — APPROVED** [review]

**Comment by cretz** [comment]
> LGTM, assuming there will be another PR once typescript is released?

Yes

File: features/schedule/basic_workflow/feature.go
Comment (mjameswh) [reply features/schedule/basic_workflow/feature.go:31]: > I can think of dozens and dozens of tests if we need to test each input form. These are the initial basic tests, not meant to validate each field.

I might have not expressed myself clearly here... I totally agree that testing tons of inputs on schedules in features tests would just be insane and would not bring in much value.

My argument was rather that, from a cross SDK point of view, if we have to retain only a very minimal number of test inputs for schedule testing, then it seems to me that there would be more value in testing some structured calandar input (because it is more likely to expose divergence between SDKs) rather than the rather trivial interval format. The specific calandar I proposed is functionally equivalent to the "every 2 seconds" interval, which means that testing methodology remains the same.

But yes, that can definitely be a distinct test... Your call.

Comment (cretz) [reply features/schedule/basic_workflow/feature.go:31]: Hrmm, I was going for testing how a user might write "every X interval" schedule.

**Pull request merged**
